src/main/java/com/notably/logic/parser/DeleteCommandParser.java

```diff
@@ -1,27 +1,50 @@
 package com.notably.logic.parser;
 
-import com.notably.commons.core.index.Index;
+import static com.notably.logic.parser.CliSyntax.PREFIX_TITLE;
+
+import java.util.stream.Stream;
+
+import com.notably.commons.core.path.RelativePath;
+import com.notably.commons.core.path.exceptions.InvalidPathException;
 import com.notably.logic.commands.DeleteCommand;
 import com.notably.logic.parser.exceptions.ParseException;
 
 /**
  * Parses input arguments and creates a new DeleteCommand object
  */
-public class DeleteCommandParser implements Parser<DeleteCommand> {
+public class DeleteCommandParser implements CommandParser<DeleteCommand> {
 
     /**
+     * TODO: integrate with CorrectionEngine.
      * Parses the given {@code String} of arguments in the context of the DeleteCommand
      * and returns a DeleteCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TITLE);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_TITLE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format("Invalid Command"));
+        }
+
+        String title = argMultimap.getValue(PREFIX_TITLE).get();
+
         try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteCommand(index);
-        } catch (ParseException pe) {
-            throw new ParseException(
-                    "MESSAGE_INVALID_COMMAND_FORMAT", pe);
+            return new DeleteCommand(RelativePath.fromString(title));
+        } catch (InvalidPathException ex) {
+            throw new ParseException(args);
         }
     }
 
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
If you search it up, it seems that this method is duplicated in 3 files. Perhaps it's time to extract move these duplications into a separate file/move it under a class?

Reply:
Yes i should definitely make this a static method in either the `NotablyParser` or `ArgumentMultiMap` class.

Reply:
👍